pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -371,25 +411,25 @@ func injectAutoInstruConfig(pod *corev1.Pod, libsToInject []libInfo) error {
 		err := injectLibInitContainer(pod, image, lang)
 		if err != nil {
 			langStr := string(lang)
-			metrics.LibInjectionErrors.Inc(langStr)
-			metrics.MutationErrors.Inc(metrics.LibInjectionMutationType, "cannot inject init container", langStr)
+			metrics.LibInjectionErrors.Inc(langStr, strconv.FormatBool(autoDetected))
+			metrics.MutationErrors.Inc(metrics.LibInjectionMutationType, "cannot inject init container", langStr, strconv.FormatBool(autoDetected))
 			lastError = err
 			log.Errorf("Cannot inject init container into pod %s: %s", podString(pod), err)
 		}
 		err = injectLibConfig(pod, lang)
```

Comment:
If the first error happens, should we do this?

Reply:
TBH this is more related to the APM part. To me, it seems like we should not inject library configuration (env variables) if we had failed to inject the library init container. But I am not sure about the author's motivation to have the current behaviour. (cc @ahmed-mez)